packages/moonstone/VirtualList/VirtualListBase.js

```diff
@@ -375,13 +398,14 @@ const VirtualListBaseFactory = (type) => {
 
 		jumpToSpottableItem = (keyCode, target) => {
 			const
-				{cbScrollTo, data, dataSize} = this.uiRef.props,
+				{isItemDisabled} = this.props,
+				{cbScrollTo, dataSize} = this.uiRef.props,
 				{firstIndex, numOfItems} = this.uiRef.state,
 				{isPrimaryDirectionVertical} = this.uiRef,
 				rtl = this.props.rtl,
 				currentIndex = Number.parseInt(target.getAttribute(dataIndexAttribute));
 
-			if (!data || !Array.isArray(data) || !data[currentIndex] || data[currentIndex].disabled) {
+			if (!isItemDisabled || isItemDisabled(currentIndex)) {
```

Comment:
I think I had misunderstood this code.  Some in-line comments might help clarify the intent of this code.  If I understand it correctly now, if there is no function for checking disabled or the current item is disabled, then allow Spotlight to handle the spotting directly.  Otherwise, we need to check to see if there are disabled items above or below (depending on direction) to figure out the next spottable item.  If this is the case, then I think this code is fine.

Reply:
You understand it correctly. That's the way how the `Virtuallist` manages spotlight with disabled items. I added the in-line comment to describe what is the mean of the condition

```
if (
	!isItemDisabled || // It is the case that there is no disabled items in a list.
	isItemDisabled(currentIndex) // If the currnet index item is disabled, it means that all items in a list are disabled.
)
```